### PR TITLE
ARROW-6919: [Python] Expose more builders in Cython

### PR DESCRIPTION
**Pull request by xhochy**

### CODE REVIEW

File: python/pyarrow/includes/libarrow.pxd [hunk 1]
```diff
@@ -1365,7 +1433,6 @@ cdef extern from "arrow/python/api.h" namespace "arrow::py":
 
 cdef extern from "arrow/python/api.h" namespace "arrow::py" nogil:
     shared_ptr[CDataType] GetPrimitiveType(Type type)
-    shared_ptr[CDataType] GetTimestampType(TimeUnit unit)
```
Comment (xhochy) [inline python/pyarrow/includes/libarrow.pxd]: This was still referenced in Cython but removed from the C++ sources.

**Review by wesm — APPROVED** [review]
+1.

**Comment by wesm** [comment]
Do you need this change in 0.15.1?

**Comment by xhochy** [comment]
No. Only my previous Cython PR needs to be in 0.15.1 but that is marked as such in JIRA.

**Pull request closed without merging**